src/platform/macos/mod.rs

```diff
@@ -427,14 +427,13 @@ impl OsIpcSender {
             // Zero out the last word for paranoia's sake.
             *((message as *mut u8).offset(size as isize - 4) as *mut u32) = 0;
 
-            let data_dest = shared_memory_descriptor_dest as *mut u8;
-            ptr::copy_nonoverlapping(data.as_ptr(), data_dest, data.len());
+            let mut data_dest = shared_memory_descriptor_dest as *mut u8;
+            let data_size = data.len();
+            let data_size_dest = data_dest as *mut usize;
+            *data_size_dest = data_size;
 
-            let mut ptr = message as *const u32;
-            let end = (message as *const u8).offset(size as isize) as *const u32;
-            while ptr < end {
-                ptr = ptr.offset(1);
-            }
+            data_dest = data_dest.offset(mem::size_of::<usize>() as isize);
+            ptr::copy_nonoverlapping(data.as_ptr(), data_dest, data_size);
```

Comment:
I can't see the message size being adjusted anywhere for the extra space taken up by the size header -- am I missing something?...

(This kind of pointer manipulation is so damn fragile -- it's actually worse than in C :-( Can't really blame you for that though I guess: probably not much point trying to use slices for this particular bit, without rewriting the rest of function in a more defensive way too...)